### PR TITLE
feat: Introducing Error/Success Messages without Redirection

### LINKED ISSUE #3: Implement `jsonWith` utility functions
Hi, 
Thanks you for this great library. 
I was thinking if you'd be up for adding JSON utility classes just like you did for redirect. 

The use-case is that sometimes you don't wan't to redirect, just send back some data and show a toast. 

That should be fairly easy to implement.

What do you think?

### PR DESCRIPTION
**Pull request by Erij-Maherzia-BEN-BRAHIM**
Fixes #3 #4

# Description

This pull request introduces a significant enhancement to the existing functionality of the library. The core feature of the library was initially centered around displaying message toasts with redirection. However, with this contribution, we've expanded the capabilities to provide users with the flexibility to showcase error or success messages without triggering a redirection.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [ ] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/test-apps/testing-app/app/routes/without-redirection.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { cssBundleHref } from "@remix-run/css-bundle";
+import { DataFunctionArgs, type LinksFunction } from "@remix-run/node";
+import { Form, useActionData } from "@remix-run/react";
+import { useEffect } from "react";
+import { displayErrorMessage, displaySuccessMessage } from "../../../../index";
```
Comment (AlemTuzlak) [inline src/test-apps/testing-app/app/routes/without-redirection.tsx:5]: you can change the import to remix-toast

File: src/index.ts [hunk 2]
```diff
@@ -36,6 +36,40 @@ async function redirectWithFlash(url: string, flash: FlashSessionValues, init?:
...
 }
 
+/**
+ * Helper method used to display a success toast notification without redirection
+ *
+ * It need to be awaited
+ * @param successMessage Message to be shown as success
+ * @returns Returns successMessage and a toast cookie set
+ */
+export async function displaySuccessMessage(successMessage: string) {
```
Comment (AlemTuzlak) [inline src/index.ts:46]: I would not do it like this, I would accept an object generic as the first argument and the toast: toastMessage as the second, and init?: ResponseInit as the third so the signature is identical to the normal ones, and then return:
```jsx
return json(
    { ...passedInData, toast },
    {
      headers: await flashMessage({
        toast: { message: successMessage, type: "success" },
      }),
    },
```
Also I would call it `jsonWithSuccess`

**Review by AlemTuzlak — CHANGES_REQUESTED** [review]
Thank you for the PR! I left a few comments on the PR, would you willing to change and adapt the code? I would make it more generic to suite every use-case and make it more consistent with the other utilities.

**Comment by Erij-Maherzia-BEN-BRAHIM** [comment]
Thank you @AlemTuzlak for reviewing my code. I will make the requested changes 👍🏻

File: src/index.ts [hunk 3]
```diff
@@ -36,6 +36,46 @@ async function redirectWithFlash(url: string, flash: FlashSessionValues, init?:
...
+/**
+ * Helper method used to display a success toast notification without redirection
+ *
+ * It need to be awaited
+ * @param passedInData Generic object containing the data
+ * @param toast Message to be shown as success
+ * @param init Additional response options (status code, additional headers etc)
+ * @returns Returns data and a toast cookie set
+ */
+export async function jsonWithSuccess<T>(passedInData: T, toast: string, init?: ResponseInit) {
```
Comment (AlemTuzlak) [inline src/index.ts:48]: sorry for the delay didn't have time to review again, so I would do the following, I'd create an utility called `jsonWithFlash` that is not exported that has the following signature:
```ts
async function jsonWithFlash<T>(data: T, flash: FlashSessionValues, init?: ResponseInit) {
  return json(data, {
    ...init,
    headers: await flashMessage(flash, init?.headers),
  });
} 
```
Then after that create a `jsonWithToast` like so:
```ts
export function jsonWithToast<T>(data: T, toast: ToastMessage, init?: ResponseInit) {
  return jsonWithFlash(data, { toast }, init);
}
```
And from there have these utilities you have created, eg:

```ts
export function jsonWithSuccess<T>(data: T, message: string, init?: ResponseInit) {
  return jsonWithToast(data, { message, type: "success" }, init);
}
```
Just so it's consitent with the redirect methods. Also I'd add jsonWithWarning and jsonWithInfo variants. If you want I can add that on top, or would you like to? @Erij-Maherzia-BEN-BRAHIM 

Also, thank you so much for this contribution :)

Comment (Erij-Maherzia-BEN-BRAHIM) [reply src/index.ts:48]: Hi @AlemTuzlak, Thank you for your feedback and suggestions! I appreciate your time in reviewing the code. I will certainly implement the changes you've recommended to ensure consistency with the redirect methods.

**Review by AlemTuzlak — APPROVED** [review]
Awesome work, will merge and release tomorrow!

**Comment by kengru** [comment]
@AlemTuzlak @Erij-Maherzia-BEN-BRAHIM thank you both for this!

**Pull request merged**